io/subsystem/src/main/java/org/wildfly/extension/io/BufferPoolResourceDefinition.java

```diff
@@ -128,29 +124,29 @@ public void registerCapabilities(ManagementResourceRegistration resourceRegistra
         resourceRegistration.registerCapability(IO_POOL_RUNTIME_CAPABILITY);
     }
 
-    private static class BufferPoolAdd extends AbstractAddStepHandler {
+    @Override
+    public void registerAttributes(ManagementResourceRegistration resourceRegistration) {
+        resourceRegistration.registerReadOnlyAttribute(BUFFER_SIZE, new BufferReadAttributeHandler(BUFFER_SIZE));
+        resourceRegistration.registerReadOnlyAttribute(BUFFER_PER_SLICE, new BufferReadAttributeHandler(BUFFER_PER_SLICE));
+        resourceRegistration.registerReadOnlyAttribute(DIRECT_BUFFERS, new BufferReadAttributeHandler(DIRECT_BUFFERS));
+    }
+
+    private static class BufferReadAttributeHandler implements OperationStepHandler {
 
-        private BufferPoolAdd() {
-            super(BufferPoolResourceDefinition.ATTRIBUTES);
+        final SimpleAttributeDefinition definition;
+
+        public BufferReadAttributeHandler(SimpleAttributeDefinition definition) {
+            this.definition = definition;
         }
 
         @Override
-        protected void performRuntime(OperationContext context, ModelNode operation, ModelNode model) throws OperationFailedException {
-            final PathAddress address = PathAddress.pathAddress(operation.get(OP_ADDR));
-            final String name = address.getLastElement().getValue();
-            final ModelNode bufferSizeModel = BUFFER_SIZE.resolveModelAttribute(context, model);
-            final ModelNode bufferPerSliceModel = BUFFER_PER_SLICE.resolveModelAttribute(context, model);
-            final ModelNode directModel = DIRECT_BUFFERS.resolveModelAttribute(context, model);
-
-            final int bufferSize = bufferSizeModel.isDefined() ? bufferSizeModel.asInt() : defaultBufferSize;
-            final int bufferPerSlice = bufferPerSliceModel.isDefined() ? bufferPerSliceModel.asInt() : defaultBuffersPerRegion;
-            final boolean direct = directModel.isDefined() ? directModel.asBoolean() : defaultDirectBuffers;
-
-            final BufferPoolService service = new BufferPoolService(bufferSize, bufferPerSlice, direct);
-            context.getCapabilityServiceTarget().addCapability(IO_POOL_RUNTIME_CAPABILITY, service)
-                    .setInitialMode(ServiceController.Mode.ACTIVE)
-                    .install();
-
+        public void execute(OperationContext context, ModelNode operation) throws OperationFailedException {
+            if ( definition.getType() == ModelType.INT )
```

Comment:
you need here to find the name of the attribute to read and return one of DEFAULT_BUFFERS_PER_REGION, DEFAULT_BUFFER_SIZE or DEFAULT_DIRECT_BUFFERS